### DISCUSSION TITLE
Request to Add Dockerfile to Repository

### DISCUSSION THREAD

**Discussion by tmciver** [firefly-iii/firefly-iii#4631] [General]
I see that there have been some issues in the past asking about the location of the Dockerfile for the project but I see that one still does not exist.

The [Dockerfile that is currently used for building the image](https://dev.azure.com/Firefly-III/_git/MainImage?path=%2FDockerfile&version=GBmain) pulls code from Github rather than use local code.  This is not ideal for developers that would like to checkout the code and build an image.

Can a Dockerfile be added to the repo that creates an image from the local code?  Azure can be made to use such a Dockerfile and is the usual way of building a Docker image.

**Comment by JC5** [comment]
I'm not going to change the current setup. With a separate Dockerfile and support file setup, I can manage the Docker image separately from the release cycle of GitHub.

**Reply by tmciver** [reply]
OK, no problem.  That's a shame though.  I think having the release cycle separated from changes in the project repo is a
_disadvantage_.  It seems counter to a typical SDLC where the build is triggered from certain events in the repository.  How do you trigger a Docker build now, manually?

Also, as I stated above, it's developer unfriendly.  But the Dockerfile needed to build an image locally is pretty straightforward so I can create one myself.  I don't suppose you'd accept a PR for a repository-based Dockerfile while continuing to use the current one, would you?

**Reply by JC5** [reply]
Yeah pretty much. I've had a few issues in the past where i've had to rebuild *old* releases with updated Docker files. I'd rather kick off another parameterized build on Azure than hack around it some other way. The Docker image is a service, and this is how its built. It also helps me in keeping the repository "clean". That's just me I guess, but I really dislike having all of these meta files all over the place.

And no, I won't accept a PR. Not because I don't want to per se, but because it puts another maintenance burden on *me*.

> This is not ideal for developers that would like to checkout the code and build an image.

And this is my problem, because? You can check out the code, and you can build an image. If you want to do both, make the effort. There are dozens of ways to build, run and tinker with Firefly III, and this one happens to need a custom Docker file.

**Reply by viniciusgava** [reply]
@JC5 could you share the docker for data-importer?

I wanna execute the dataimport without turn on the webserver. I have some scripts that connect in the my accounts(most of them dockers - nodejs and Puppeteer) to get the CSV and once they finished the download I would like to start the data-importer container only triggering the command line  `artisan importer:import`. The idea is execute the container to import and die.

**Reply by JC5** [reply]
What do you need exactly? [This](https://docs.firefly-iii.org/data-importer/install/docker/) or [this](https://dev.azure.com/Firefly-III/_git/ImportToolImage?path=/Dockerfile)?

**Reply by MacPaille** [reply]
Fidi can run easily without the Web interface. Check the documentation for the env variable to set. If I recall well it ser WEB_SERVER at false.
It will just start, look at a folder for files to import and stop.

That's what I use with a cron task on my Synology to start my container everyday at 7PM. It will auto import all my transactions in my 3 main accounts using the Nordigen config and the others it reads from the csv. I manually update them every few days

**Reply by MacPaille** [reply]
This is the page for the automation that's what I use.
https://docs.firefly-iii.org/data-importer/usage/command_line/

**Reply by viniciusgava** [reply]
Thank you guys.
@JC5 I was looking for the Dockerfile, second link, my idea was change the command executed on the boot, instead of turn on the apache, just execute a php artisan command.

 I was looking for trigger a docker run per file, but looking the Dockerfile it seems it is gonna be a complicated because of the entrypoint. I'll start my integrations using the `WEB_SERVER=false` / `importer:auto-import /import` .
